File: 2593-find-score-of-an-array-after-marking-all-elements/2593-find-score-of-an-array-after-marking-all-elements.java

class Solution {
    public long findScore(int[] nums) {
        Set<Integer> visited=new HashSet<>();
        PriorityQueue<int[]> min_heap=new PriorityQueue<>((a,b)->
        (a[0]==b[0])?a[1]-b[1]:a[0]-b[0]);
        long ans=0;
        for(int i=0;i<nums.length;i++){
            min_heap.add(new int[]{nums[i],i});
        }
        while(!min_heap.isEmpty()){
            int[] curr=min_heap.poll();
            int curr_ele=curr[0];
            int curr_idx=curr[1];
            if(!visited.contains(curr_idx)){
                visited.add(curr_idx);
                ans+=curr_ele;
                if((curr_idx+1)>=0) visited.add(curr_idx+1);
                if((curr_idx-1)>=0) visited.add(curr_idx-1);
            }
        }
        return ans;
    }

}

File: 2593-find-score-of-an-array-after-marking-all-elements/2593-find-score-of-an-array-after-marking-all-elements.py
class Solution:
    def findScore(self, nums: List[int]) -> int:
        visited=set()
        min_heap=[]
        ans=0
        for i in range(len(nums)):
            heappush(min_heap,(nums[i],i))
        while(min_heap):
            curr=heappop(min_heap)
            curr_ele=curr[0]
            curr_idx=curr[1]
            if curr_idx not in visited:
                ans+=curr_ele
                visited.add(curr_idx)
                if (curr_idx-1)>=0:
                    visited.add(curr_idx-1)
                if(curr_idx+1)<len(nums):
                    visited.add(curr_idx+1)
        return ans
                